chrome/content/zotero/elements/itemDetails.js

```diff
@@ -0,0 +1,582 @@
+			if (!collapsible) return;
+			let splitter = collapsible.previousElementSibling;
+			if (val) {
+				collapsible.setAttribute('collapsed', 'true');
+				collapsible.removeAttribute("width");
+				collapsible.removeAttribute("height");
+				splitter.setAttribute('state', 'collapsed');
+				splitter.setAttribute('substate', 'after');
+			}
+			else {
+				collapsible.removeAttribute('collapsed');
+				splitter.setAttribute('state', '');
+				splitter.setAttribute('substate', 'after');
+			}
+			window.dispatchEvent(new Event('resize'));
+		}
+
+		get sidenav() {
+			return this._sidenav;
+		}
+
+		set sidenav(sidenav) {
+			this._sidenav = sidenav;
+			sidenav.container = this;
+		}
+
+		static get observedAttributes() {
+			return ['pinnedPane'];
+		}
+
+		init() {
+			this._container = this.querySelector('#zotero-view-item-container');
+			this._header = this.querySelector('#zotero-item-pane-header');
+			this._paneParent = this.querySelector('#zotero-view-item');
+
+			this._container.addEventListener("keypress", this._handleKeypress);
+			this._paneParent.addEventListener('scroll', this._handleContainerScroll);
+
+			this._paneHiddenOb = new MutationObserver(this._handlePaneStatus);
+			this._paneHiddenOb.observe(this._paneParent, {
+				attributes: true,
+				attributeFilter: ["hidden"],
+				subtree: true,
+			});
+			this._initIntersectionObserver();
+
+			this._unregisterID = Zotero.Notifier.registerObserver(this, ['item', 'itempane'], 'ItemDetails');
+
+			this._disableScrollHandler = false;
+			this._pinnedPaneMinScrollHeight = 0;
+
+			this._lastUpdateCustomSection = 0;
+		}
+
+		destroy() {
+			this._container.removeEventListener("keypress", this._handleKeypress);
+			this._paneParent.removeEventListener('scroll', this._handleContainerScroll);
+
+			this._paneHiddenOb.disconnect();
+			this._intersectionOb.disconnect();
+
+			Zotero.Notifier.unregisterObserver(this._unregisterID);
+		}
+
+		async render() {
+			if (!this.initialized || !this.item) {
+				return;
+			}
+			let item = this.item;
+			Zotero.debug('Viewing item');
+			this._isRendering = true;
+
+			this.renderCustomSections();
+
+			let panes = this.getPanes();
+			let pendingBoxes = [];
+			let inTrash = ZoteroPane.collectionsView.selectedTreeRow && ZoteroPane.collectionsView.selectedTreeRow.isTrash();
+			let tabType = Zotero_Tabs.selectedType;
+			for (let box of [this._header, ...panes]) {
+				box.mode = this.mode || "edit";
+				box.item = item;
+				box.inTrash = inTrash;
+				box.tabType = tabType;
+				// Render sync boxes immediately
+				if (!box.hidden && box.render) {
+					if (box.render.constructor.name == "AsyncFunction") {
+						pendingBoxes.push(box);
+					}
+					else {
+						box.render();
+					}
+				}
+			}
+
+			let pinnedPaneElem = this.getEnabledPane(this.pinnedPane);
+			let pinnedIndex = panes.indexOf(pinnedPaneElem);
+			
+			this._paneParent.style.paddingBottom = '';
+			await waitFrames(3);
```

Comment:
I guess 1 frame would be enough! We need to wait for panes to be hidden/visible here and I forget why I picked 3

Reply:
Well, just tested and 1 frame won't do - switching between attachment and regular item in the library with tags section pinned scrolls to the incorrect position. So at least for me wait for 3 frames works. I will add a comment.

Reply:
That's true but doesn't work either. I'll try to use the `hidden` as attribute instead of property in boxes to see if we can get this fixed. Probably XUL's `hidden` is the cause?

Reply:
Here are two different things - the waiting frame is to solve `Test case: pin tags section and switch between attachment and regular item`

Reply:
OK fixed in [40a56a8](https://github.com/zotero/zotero/pull/3624/commits/40a56a807176b40efa6703e9c0ecef2af9e24fd5). I forget the attachment preview has a self-hosted resize observer!